### DISCUSSION TITLE
inconsistant string methods

### DISCUSSION THREAD

**Discussion by Flinner** [vlang/v#11880] [Ideas]
Expect all_after to output nothing, there is nothing after 'ahhh'
```go
'hello world'.all_after('ahhh')
//output: 'hello world'
```
Expect find_between to output everthing, even if there is no 'ahh' or 'yay', just like `all_after`

```go
'hello world'.find_between('ahh', 'yay')
//output: nothing
```

**Comment by mcastorina** [comment]
I agree we need consistency, but I think the behavior is debatable. I would argue that both methods should return nothing.

Perhaps these methods could return `?string` to differentiate between `none` and `''`.

**Comment by dumblob** [comment]
@mcastorina please don't confuse `?string` and `string | none` - those are two wildly different things (see https://github.com/vlang/v/issues/9598 :wink:).

I guess you wanted to actually write `string | none` because there is no error involved so no error propagation (i.e. optionals) required.

And if there was an error required, then it'd become `?(string | none)` (*V* doesn't seem to support this syntax yet, so you'd need to first declare a new sum type `type MaybeString string | none; fn all_after( x string ) ?MaybeString { ... }`).

**Comment by mcastorina** [comment]
Yes, when I said `?string` the intention was `string | none` which (correct or not) is the current recommended way to return an optional value.

That is indeed a good discussion to follow, and I hope the distinction becomes more clear in future versions of V.
